lib/v8.js

```diff
@@ -42,3 +51,21 @@ exports.getHeapStatistics = function() {
 };
 
 exports.setFlagsFromString = v8binding.setFlagsFromString;
+
+exports.getHeapSpaceStatistics = function() {
+  const heapSpaceStatistics = [];
+  const buffer = heapSpaceStatisticsBuffer;
+
+  for (let i = 0; i < kNumberOfHeapSpaces; i++) {
+    const heapSpaceName = v8binding.updateHeapSpaceStatisticsArrayBuffer(i);
```

Comment:
This makes `kNumberOfHeapSpaces` runtime calls which is arguably a bit wasteful.  I'd make `heapSpaceStatisticsBuffer` big enough to hold `kNumberOfHeapSpaces` records and make a single call.